system76-power: init at 1.1.16

jwoudenberg opened the pull request:
###### Motivation for this change

NixOS already comes bundled with support for system76 firmware. It was great to see this when I bought a new system76 machine, and it worked out of the box.

I did notice when booting into NixOS for the first time that the fans were significantly louder than they were in the preinstalled PopOS installation. After some investigation I found this was because I wasn't running the system76-power daemon. This PR adds support for that daemon.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jwoudenberg committed:
system76-power: init at 1.1.16

r-rmcgibbo commented:
Result of `nixpkgs-review pr 126656` at 0bda5136 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>linuxPackages.system76-power (linuxPackages-libre.system76-power ,linuxPackages_4_14.system76-power ,linuxPackages_4_19.system76-power ,linuxPackages_4_4.system76-power ,linuxPackages_4_9.system76-power ,linuxPackages_5_10.system76-power ,linuxPackages_5_12.system76-power ,linuxPackages_5_4.system76-power ,linuxPackages_hardened.system76-power ,linuxPackages_hardkernel_4_14.system76-power ,linuxPackages_hardkernel_latest.system76-power ,linuxPackages_latest-libre.system76-power ,linuxPackages_latest.system76-power ,linuxPackages_latest_hardened.system76-power ,linuxPackages_lqx.system76-power ,linuxPackages_testing_bcachefs.system76-power ,linuxPackages_xanmod.system76-power ,linuxPackages_zen.system76-power)</li>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/system76-power/default.nix:29:5:
```
   |
29 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

jwoudenberg commented:
Pinging @shlevy and @khumba as potentially interested parties (because you worked on this module before, don't know if you consider yourself maintainer though!)

I'd be happy to help out as maintainer for any of the system76 packages and modules btw!

khumba reviewed (commented):
Hi @jwoudenberg, thanks for adding this service!  The code looks good, I've just got a few comments.  I've built the package but haven't tested a system with this enabled yet.

jwoudenberg commented on pkgs/os-specific/linux/system76-power/default.nix:
```diff
@@ -0,0 +1,33 @@
+  cargoSha256 = "sha256-YqGr6Qou73gBDgz+S4oNhPQfrv7rkys+xZkgMCQca1M=";
+
+  postInstall = ''
+    install -D -m -0644 data/system76-power.conf $out/etc/dbus-1/system.d/system76-power.conf
```
Oh, good catch. Gotta admit I borrowed this bit from elsewhere and didn't look as closely as I should have.
I can't find any documentation supporting a minus in that position, so I think it's a mistake too.

Based on a quick experiment it doesn't look like it's setting `000` permissions though:

```
$ touch foo
$ install -D -m -0644 foo bar
$ ls -l | grep bar
-rwsrwsrwt    1 root     root             0 Jun 12 18:53 bar
```

Will push a fix!

jwoudenberg commented:
Hi @khumba. Thank you for your review! I made the changes you proposed, but am away from my NixOS box over the weekend. I'll let you know when I've had a chance to check them on my system76 thelio machine, probably tomorrow evening.

SuperSandro2000 commented on pkgs/os-specific/linux/system76-power/default.nix:
```diff
@@ -0,0 +1,33 @@
+{ pkg-config, libusb1, dbus, lib, rustPlatform, fetchFromGitHub }:
+
+let
+  version = "1.1.16";
+  sha256 = "sha256-OtrhvUkNNgg6KlrqjwiBKL4zuQZBWevb0xgtSlEW2rQ=";
+in rustPlatform.buildRustPackage {
+  pname = "system76-power";
+  version = version;
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = "system76-power";
+    rev = version;
+    sha256 = sha256;
```
```suggestion
rustPlatform.buildRustPackage rec {
  pname = "system76-power";
  version = "1.1.16";

  src = fetchFromGitHub {
    owner = "pop-os";
    repo = "system76-power";
    rev = version;
    sha256 = "sha256-OtrhvUkNNgg6KlrqjwiBKL4zuQZBWevb0xgtSlEW2rQ=";
```

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -5140,6 +5140,12 @@
     email = "jwilberding@afiniate.com";
     name = "Jordan Wilberding";
   };
+  jwoudenberg = {
```
Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

shlevy commented:
Awesome!

jwoudenberg committed:
maintainers: add jwoudenberg

jwoudenberg commented:
I tested this on my system76 machine and all seems to work:
- `systemctl status system76-power` seems happy
- `system76-power --help` seems happy

@khumba and @SuperSandro2000, I believe I addressed all your review comments. I'd super appreciate it if you could take another look!

khumba reviewed (approved):
Looks good to me, thanks @jwoudenberg!  I don't have commit access so someone else will have to merge.  :star: :rocket:

(Also just a minor note that commits should be in order if possible so that each commit will build, so it's better the maintainer commit comes first.)

aanderse reviewed (approved):
Module LGTM :+1: Nice work.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/thelio-constant-fan-noise-working-great-with-popos/15185/2
